src/go/k8s/apis/redpanda/v1alpha1/console_types.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ConsoleSpec defines the desired state of Console
+type ConsoleSpec struct {
+	// +optional
+	// +kubebuilder:default=console
+	MetricsNamespace string `json:"metricsNamespace"`
+
+	// +optional
+	// +kubebuilder:default=true
+	ServeFrontend bool `json:"serveFrontend"`
+
+	// +optional
+	Server Server `json:"server"`
+
+	Schema        Schema                 `json:"schema"`
+	ClusterKeyRef corev1.ObjectReference `json:"clusterKeyRef"`
+	Deployment    Deployment             `json:"deployment"`
+	Connect       Connect                `json:"connect"`
+}
+
+// Server is the Console app HTTP server config
+type Server struct {
+	// +kubebuilder:default="30s"
+	ServerGracefulShutdownTimeout string `json:"gracefulShutdownTimeout,omitempty" yaml:"gracefulShutdownTimeout,omitempty"`
```

Comment:
nit: should we write what is the format of this? Or is this something standard? What do other CRDs do to define timeouts?

Reply:
:+1: There is a `metav1.Duration` type from apimachinery, I'll use that here [REF](https://github.com/redpanda-data/redpanda/pull/5758/commits/8501c85e4568aeaf11bd855f6fe1fb9481c735dc)

Reply:
Marking as resolved [REF](https://github.com/redpanda-data/redpanda/pull/5758/commits/f01191d4eb0b41942e6c2a53cb05076f098ac7ce) The ref above might have been force pushed